src/MapUtil/MapUtil.spec.ts

```diff
@@ -857,4 +858,48 @@ describe('MapUtil', () => {
     expect(MapUtil.getLayerByNameParam(map, testFeatureType)?.getVisible()).toBe(false);
   });
 
+  describe('#calculateScaleAndCenterForExtent', () => {
+    it('returns scale and center for a valid extent', () => {
+      const extent = [0, 0, 100, 100];
+      const view = new OlView({
+        projection: 'EPSG:3857',
+        resolutions: [1, 0.5, 0.25]
+      });
+      const olMap = new OlMap({view});
+      view.setResolution(1);
+      view.setCenter([50, 50]);
+
+      const result = MapUtil.calculateScaleAndCenterForExtent(olMap, extent);
+
+      expect(result).toBeDefined();
+      expect(result?.center).toEqual([50, 50]);
```

Comment:
The equality test might work here, but in general it would be better to compare floating values with a delta since `0.1 + 0.2 !== 0.3`.